src/e2ir.c

```diff
@@ -1875,6 +1875,68 @@ elem *toElem(Expression *e, IRState *irs)
             return e;
         }
 
+        elem *toElemBinAssign(BinAssignExp *be, int op)
+        {
+            //printf("toElemBinAssign() '%s'\n", be->toChars());
+
+            Type *tb1 = be->e1->type->toBasetype();
+            Type *tb2 = be->e2->type->toBasetype();
+
+            assert(!((tb1->ty == Tarray || tb1->ty == Tsarray ||
+                      tb2->ty == Tarray || tb2->ty == Tsarray) &&
+                     tb2->ty != Tvoid &&
+                     op != OPeq && op != OPandand && op != OPoror));
+
+            tym_t tym = totym(be->type);
+
+            elem *el;
+            elem *ev;
+            if (be->e1->op == TOKcast)
```

Comment:
Why is it valid for the lhs to be a CastExp?  I didn't think this was a valid ast...  Which expressions produce this, and is it a frontend bug that they get this far?